src/server/LoginServer/DBProcess.cpp

```diff
@@ -200,146 +166,143 @@ int CDBProcess::MgameLogin(const char * id, const char * pwd) {
             }
         }
 
-        SQLFreeHandle((SQLSMALLINT)SQL_HANDLE_STMT, hstmt);
+        SQLFreeHandle((SQLSMALLINT)SQL_HANDLE_STMT, hStmt);
     }
 
     return sParmRet;
 }
 
-BOOL CDBProcess::InsertVersion(int version, const char * filename, const char * compname, int historyversion) {
-    SQLHSTMT  hstmt = NULL;
-    SQLRETURN retcode;
-    TCHAR     szSQL[1024];
-    memset(szSQL, 0x00, 1024);
+BOOL CDBProcess::InsertVersion(short sVersion, const fs::path & fsFile, const char * szCompName,
+                               short sHistoryVersion) {
     BOOL retvalue = TRUE;
 
-    wsprintf(
+    // convert narrow to wide
+    std::wstring szTableName(m_pMain->m_TableName, m_pMain->m_TableName + strlen(m_pMain->m_TableName));
+    std::wstring szCompName2(szCompName, szCompName + strlen(szCompName));
+
+    wchar_t szSQL[1024]{};
+    swprintf_s(
         szSQL,
-        TEXT(
-            "INSERT INTO %s (sVersion, strFileName, strCompressName, sHistoryVersion) VALUES (%d, \'%s\', \'%s\', %d)"),
-        m_pMain->m_TableName, version, filename, compname, historyversion);
-
-    retcode = SQLAllocHandle((SQLSMALLINT)SQL_HANDLE_STMT, m_VersionDB.m_hdbc, &hstmt);
-    if (retcode == SQL_SUCCESS) {
-        retcode = SQLExecDirect(hstmt, (unsigned char *)szSQL, SQL_NTS);
-        if (retcode != SQL_SUCCESS && retcode != SQL_SUCCESS_WITH_INFO) {
-            DisplayErrorMsg(hstmt);
+        L"INSERT INTO %s (sVersion, strFileName, strCompressName, sHistoryVersion) VALUES (%hd, N'%s', '%s', %hd)",
+        szTableName.c_str(), sVersion, fsFile.c_str(), szCompName2.c_str(), sHistoryVersion);
```

Comment:
Interesting choice using a Unicode string for `strFileName`. You’ll likely need to update the database for this to work in this PR. One issue is that you’re calling `std::filesystem::path::c_str()` directly instead of using `std::filesystem::path::wstring()` followed by `std::wstring::c_str()` for better portability. Note that on Windows, `std::filesystem::path::c_str()` returns `wchar_t`, while on POSIX systems it returns `char` due to the default UTF-8 encoding, whereas Windows often defaults to ANSI encoding. Microsoft has historically lagged in supporting encoding compatibility.

Good news: starting with MSVC 14.00 (Visual Studio 2015), Microsoft introduced the [`/utf-8` compiler flag](https://learn.microsoft.com/en-us/cpp/build/reference/utf-8-set-source-and-executable-character-sets-to-utf-8), allowing strings to behave similarly to POSIX by treating them as UTF-8 instead of ANSI. I recommend enabling this flag in your project settings, especially since you’re using `std::filesystem` here. For example:
```cpp
const std::filesystem::path p(u8"要らない.txt");
std::ofstream(p) << "File contents";
```

With `u8` string literals, this will work cross-platform and write the file correctly. Without `u8`:
```cpp
const std::filesystem::path p("要らない.txt");
std::ofstream(p) << "File contents";
```

On Windows, `ofstream` will fail to write the file because it treats the string as ANSI, corrupting the unicode characters. Compiling with `/utf-8` resolves this, so you can avoid the complexities of C++20 Unicode types (`char8_t`, `char16_t`, `char32_t`, or `std::u*string`).

I mention this because, with MSSQL, we could adopt a similar approach to reduce storage size by setting the database collation to [`_UTF8`](https://learn.microsoft.com/en-us/sql/relational-databases/collations/collation-and-unicode-support?view=sql-server-ver16), which avoids the complexity of `nvarchar` and `nchar` types (UTF-16 encoded). Note that this only works from MSSQL 2019 onward, so if you choose this approach, your application must be UTF-8 aware and treat `varchar` and `char` types as UTF-8 encoded. This lets you avoid `std::wstring`s and `nvarchar` columns and generally makes the application more portable.

That said, your current approach with `wchar_t` and `nvarchar`/`nchar` is also correct and portable, especially for compatibility with older database engines. But if you don’t need that support, a full UTF-8 approach on both the application and database sides might be a better choice.

Reply:
> You’ll likely need to update the database for this to work in this PR.

Indeed, I forgot adding this in the PR. Will do soon.

> One issue is that you’re calling `std::filesystem::path::c_str()` directly instead of using `std::filesystem::path::wstring()` followed by `std::wstring::c_str()` for better portability. Note that on Windows, `std::filesystem::path::c_str()` returns `wchar_t`, while on POSIX systems it returns `char` due to the default UTF-8 encoding, whereas Windows often defaults to ANSI encoding.

Great feedback. Funnily enough I know this, but somehow missed this this.

About adding `/utf-8` compiler flag and `_UTF8` collation, this is amazing! 🤯 I always hated Microsoft for making string literals as ANSI strings instead of UTF8. The compiler flag `/utf-8` I'll definitely add, but updating the collation might be tricky, as I actually like the fact that we have backward compatability with older database engines and if that means we set the minimum version to MSSQL 2019, I'd have to ask the community using this version, like making a vote or something. I know some people still use MSSQL 2008 due to the official 1298 server files.

Reply:
@xGuTeK, @srmeier, @UTengine, @ko4life-net/steveteam, what's your thoughts about making MSSQL2019 the minimum version requirements and dropping backward compatability due to `UTF8` newly added collation? Are any of you still using older MSSQL versions?

Reply:
From the docs: https://learn.microsoft.com/en-us/sql/relational-databases/collations/collation-and-unicode-support?view=sql-server-ver16#utf8

> SQL Server 2019 (15.x) introduces full support for the widely used UTF-8 character encoding as an import or export encoding, and as database-level or column-level collation for string data. UTF-8 is allowed in the char and varchar data types, and it's enabled when you create or change an object's collation to a collation that has a UTF8 suffix. One example is changing LATIN1_GENERAL_100_CI_AS_SC to LATIN1_GENERAL_100_CI_AS_SC_UTF8.

Reply:
For me no problem. New version = better security and new features ;)

Reply:
I just learned that there is a beta feature on Windows 11 to set System Locale to use UTF-8 globally. This is interesting, but I havne't tested with it enough to tell if it is stable or messed up with other things.
![image](https://github.com/user-attachments/assets/7a0935a5-8671-47f7-ac33-c42599bcd505)

Reply:
Implemented `/utf-8` compiler flags and utf8 database collation `Latin1_General_100_CS_AS_SC_UTF8`. Had to rework the database a bit to make this happen, which was done in here:
- https://github.com/ko4life-net/ko-db/pull/37
- https://github.com/ko4life-net/ko-db/pull/39

Also I reverted that change with nvarchar column to use varchar considering we now use UTF8 collation: https://github.com/ko4life-net/ko/pull/260/commits/57bf02930fff2ff25e741760261a2f8873ff12c8

Awesome feedback! 🚀